services/distribution/src/main/java/app/coronawarn/server/services/distribution/statistics/local/LocalStatisticsToProtobufMapping.java

```diff
@@ -0,0 +1,228 @@
+
+  private final StatisticJsonFileLoader jsonFileLoader;
+  private final LocalStatisticsDownloadService localStatisticsDownloadService;
+  private final RegionMappingConfig regionMappingConfig;
+
+  /**
+   * Process the JSON file provided by TSI and map the it to Local Statistics protobuf object.
+   *
+   * @param jsonFileLoader - Loader of the file from the system
+   * @param localStatisticsDownloadService - Local Statistics Download Service for keeping track of ETags
+   */
+  public LocalStatisticsToProtobufMapping(StatisticJsonFileLoader jsonFileLoader,
+      LocalStatisticsDownloadService localStatisticsDownloadService,
+      RegionMappingConfig regionMappingConfig) {
+    this.jsonFileLoader = jsonFileLoader;
+    this.localStatisticsDownloadService = localStatisticsDownloadService;
+    this.regionMappingConfig = regionMappingConfig;
+  }
+
+  /**
+   * Create protobuf local statistic map from raw JSON local statistics.
+   * Local statistics has to be uploaded on CDN in 7 packages. Each entry on the map represents 1 package.
+   * Packages of local statistics contains data about 1 up to 4 federal states and all their administrative units.
+   * For more info related to the grouping of the federal states into packages please look into
+   * https://github.com/corona-warn-app/cwa-app-tech-spec/blob/42e9e4f3c588cd2fd283f904e9f0ccd53a2b83d0/
+   * docs/spec/statistics.md#populate-local-statistical-data
+   *
+   * @return map containing local statistics grouped by archive id.
+   */
+  @Bean
+  public Map<Integer, LocalStatistics> constructProtobufLocalStatistics() {
+    Map<Integer, LocalStatistics> localStatisticsMap = new HashMap<>();
+
+    try {
+      Optional<JsonFile> optionalFile = this.getFile();
+
+      if (optionalFile.isEmpty()) {
+        logger.warn("Stats file is already updated to the latest version. Skipping generation.");
+        return Collections.emptyMap();
+      } else {
+        JsonFile file = optionalFile.get();
+        this.updateETag(optionalFile.get().getETag());
+
+        List<LocalStatisticsJsonStringObject> onePerProvinceStatistics = deserializeAndValidate(file);
+
+        onePerProvinceStatistics.forEach(localStatisticsJsonStringObject -> {
+          if (localStatisticsJsonStringObject.getProvinceCode() != null) {
+            int provinceCode = Integer.parseInt(localStatisticsJsonStringObject.getProvinceCode());
+
+            if (isFederalState(localStatisticsJsonStringObject.getProvinceCode())) {
+              fillLocalStatisticsFederalStatesGroupMap(
+                  localStatisticsMap,
+                  provinceCode,
+                  federalStateSupplier(provinceCode, localStatisticsJsonStringObject),
+                  federalStateEnhancer(provinceCode, localStatisticsJsonStringObject)
+              );
+            } else {
+
+              Optional<String> federalStateCodeOptional = findFederalStateByProvinceCode(
```

Comment:
`findFederalStateByProvinceCode(int i) { return i % 1000; } `

Reply:
hmm I agree with this approach, it is more elegant but I think there should be division operation not modulo, right ?
15334 / 1000 = 15 ; 8345 / 1000 = 8.
15334 % 1000 = 334 ; 8345 / 1000 = 345